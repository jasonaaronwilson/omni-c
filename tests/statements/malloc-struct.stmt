value_hashtable_t* result = malloc_struct(value_hashtable_t);
