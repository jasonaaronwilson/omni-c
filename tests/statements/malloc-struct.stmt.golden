{
    tag = PARSE_NODE_VARIABLE_DEFINITION
    name = result
    type = {
        tag = PARSE_NODE_TYPE
        type_node_kind = TYPE_NODE_KIND_POINTER
        type_args = [
            {
                tag = PARSE_NODE_TYPE
                type_node_kind = TYPE_NODE_KIND_TYPENAME
                type_name = value_hashtable_t
                type_args = [
                ]
            }
        ]
    }
    value = {
        tag = PARSE_NODE_CALL
        function = {
            tag = PARSE_NODE_IDENTIFIER
            token = malloc_struct
        }
        args = [
            {
                tag = PARSE_NODE_IDENTIFIER
                token = value_hashtable_t
            }
        ]
    }
}

// C Output
value_hashtable_t* result = malloc_struct(value_hashtable_t);

